src/python/zquantum/core/interfaces/backend_test.py

```diff
@@ -117,6 +147,175 @@ def test_get_bitstring_distribution(self, backend):
         assert bitstring_distribution.distribution_dict["111"] > 1 / 3
 
 
+class QuantumBackendGatesTests:
+    @pytest.mark.parametrize(
+        "initial_gate,tested_gate,target_values",
+        one_qubit_non_parametric_gates_exp_vals_test_set,
+    )
+    def test_one_qubit_non_parametric_gates_using_expectation_values(
+        self, backend_for_gates_test, initial_gate, tested_gate, target_values
+    ):
+
+        if backend_for_gates_test.n_samples is None:
+            pytest.xfail(
+                "This test won't work for simulators without sampling, it should be covered by a test in QuantumSimulatorTests."
+            )
+
+        # Given
+        qubit_list = [Qubit(0)]
+        gate_1 = Gate(initial_gate, qubits=qubit_list)
+        gate_2 = Gate(tested_gate, qubits=qubit_list)
+
+        circuit = Circuit()
+        circuit.gates = [gate_1, gate_2]
+        operators = [
+            QubitOperator("[]"),
+            QubitOperator("[X0]"),
+            QubitOperator("[Y0]"),
+            QubitOperator("[Z0]"),
+        ]
+
+        sigma = 1 / np.sqrt(backend_for_gates_test.n_samples)
+
+        for i, operator in enumerate(operators):
+            # When
+            estimator = BasicEstimator()
+            expectation_value = estimator.get_estimated_expectation_values(
+                backend_for_gates_test,
+                circuit,
+                operator,
+            ).values[0]
+
+            # Then
+            assert expectation_value == pytest.approx(target_values[i], abs=sigma * 3)
```

Comment:
This may be nit-picky, so feel free to ignore, but shouldn't we check the expectation values of the individual operators? Getting the sum of `I`, `X`, `Y`, and `Z` isn't as precise as measuring and comparing them separately

Reply:
We do check for every single operator. Check is happening inside a loop and we check for every single operator separately.

Reply:
Whoops, my bad. Read this incorrectly :)